### ISSUE TITLE
type: class expression that extends passed argument of generic TypeConstructor is reported incorrectly by typeChecker as intersection of types

### ISSUE THREAD

**Issue by armano2** [microsoft/TypeScript#49116]
# Bug Report

### 🔎 Search Terms

- type
- generic extends
- class expression

### 🕗 Version & Regression Information

- This allways reported incorrect data tested between versions 3.3.x and 4.6.2

### ⏯ Playground Link

[Playground Link](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAYQDYEMDOmDqMoAWAcugLYCmAPACoBCW5i5AHlOWACaaJjkDuiABQA6UegBOAc0wAuROjABPANoBdAJSIAvAD5EAbwC+OwfUzk5dBpv0BYAFABIceSghxSCBmxNW7LohmjHb2Do6OAA7iMABu6GyIAPpgZBaImFDRYJIA3A5hkSAARqgwEIhsGYJxqCBpGVmSNgWOhDCYwsmp2og1dXlOjoZhw-ajQA)

[Typescript Eslint](https://typescript-eslint.io/play/#ts=4.6.4&sourceType=module&code=GYVwdgxgLglg9mABAYQDYEMDOmDqMoAWAcugLYCmAPACoBCW5i5AHlOWACaaJjkDuiABQA6UegBOAc0wAuROjABPANoBdAJSIAvAD5EAbwC+OwfUzk5dBpv0BYAFABIceSghxSCBmxNW7LohmjHb2Do6OAA7iMABu6GyIAPpgZBaImFDRYJIA3A5hkSAARqgwEIhsGYJxqCBpGVmSNgWOhDCYwsmp2og1dXlOjoZhw-aGQA&rules=N4IgAgLgngDgpgZwMYCcCWMIFpEBs0B2EA9DCnAGZwpbkCGAJgPYG5QgBcI1KTKIAXyA&tsConfig=N4XyA)

[Ast Explorer](https://ts-ast-viewer.com/#code/GYVwdgxgLglg9mABAYQDYEMDOmDqMoAWAcugLYCmAPACoBCW5i5AHlOWACaaJjkDuiABQA6UegBOAc0wAuROjABPANoBdAJSIAvAD5EAbwC+OwfUzk5dBpv0AoAJDjyUEOKQQM2Jq3ZdEZxjsHewAHcRgAN3Q2RAB9MDILREwocLBJAG5bYJCQACNUGAhENhTBKNQQJJS0yRtg+0IYTGF4xO1ECqqs+3tDB37DIA)

### 💻 Code

```ts
function ClassWithName<TBase extends new (...args: any[]) => {}>(Base: TBase) {
	return class A extends Base {
	}
}
```

### 🙁 Actual behavior

Type of ClassExpression that extends generic not default TypeConstructor is incorrectly reported as Intersection without symbol set

![image](https://user-images.githubusercontent.com/625469/168471215-18fa8582-9def-4bc6-b857-ed12279c387c.png)

### 🙂 Expected behavior

i expect to receive object with type TypeObject that has Object typeFlag set and properly defined symbol

This issue is present only if we extend class by other class thatr is provided by arg of function `<TBase extends new (...args: any[]) => {}>(Base: TBase)` and only if its a generic (template)

ref: https://github.com/typescript-eslint/typescript-eslint/pull/4974

**Comment by andrewbranch** [comment]
https://github.com/microsoft/TypeScript/pull/13743
